LockerResolverTest throws null point exception when tidb cluster is not present.

```
[INFO] Running com.pingcap.tikv.txn.LockResolverTest
[ERROR] Tests run: 6, Failures: 0, Errors: 6, Skipped: 0, Time elapsed: 0.004 s <<< FAILURE! - in com.pingcap.tikv.txn.LockResolverTest
[ERROR] RCTest(com.pingcap.tikv.txn.LockResolverTest)  Time elapsed: 0.003 s  <<< ERROR!
java.lang.NullPointerException
	at com.pingcap.tikv.txn.LockResolverTest.RCTest(LockResolverTest.java:443)

[ERROR] SITest(com.pingcap.tikv.txn.LockResolverTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException
	at com.pingcap.tikv.txn.LockResolverTest.SITest(LockResolverTest.java:406)

[ERROR] cleanLockTest(com.pingcap.tikv.txn.LockResolverTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException
	at com.pingcap.tikv.txn.LockResolverTest.cleanLockTest(LockResolverTest.java:310)

[ERROR] getRCTest(com.pingcap.tikv.txn.LockResolverTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException
	at com.pingcap.tikv.txn.LockResolverTest.putAlphabet(LockResolverTest.java:227)
	at com.pingcap.tikv.txn.LockResolverTest.getRCTest(LockResolverTest.java:299)

[ERROR] getSITest(com.pingcap.tikv.txn.LockResolverTest)  Time elapsed: 0 s  <<< ERROR!
java.lang.NullPointerException
	at com.pingcap.tikv.txn.LockResolverTest.putAlphabet(LockResolverTest.java:227)
	at com.pingcap.tikv.txn.LockResolverTest.getSITest(LockResolverTest.java:274)

[ERROR] txnStatusTest(com.pingcap.tikv.txn.LockResolverTest)  Time elapsed: 0.001 s  <<< ERROR!
java.lang.NullPointerException
	at com.pingcap.tikv.txn.LockResolverTest.txnStatusTest(LockResolverTest.java:358)
```

We certaninly can improve the error info.

--- a/tikv-client/src/test/java/com/pingcap/tikv/predicates/ScanAnalyzerTest.java
+++ b/tikv-client/src/test/java/com/pingcap/tikv/predicates/ScanAnalyzerTest.java
@@ -201,7 +201,7 @@ public class ScanAnalyzerTest {
   public void logicalBinaryExpressionToIndexRangesTest() {
     TiTableInfo table = createTableWithIndex(6, 5);
     TiIndexInfo index = table.getIndices().get(0);
-    
+
     TypedKey zero = TypedKey.toTypedKey(0, IntegerType.INT);
     TypedKey one = TypedKey.toTypedKey(1, IntegerType.INT);
     TypedKey two = TypedKey.toTypedKey(2, IntegerType.INT);

--- a/tikv-client/src/test/java/com/pingcap/tikv/txn/LockResolverTest.java
+++ b/tikv-client/src/test/java/com/pingcap/tikv/txn/LockResolverTest.java
@@ -62,6 +62,7 @@ public class LockResolverTest {
   private static final int DefaultTTL = 10;
   private BackOffer backOffer = ConcreteBackOffer.newCustomBackOff(1000);
   private ReadOnlyPDClient pdClient;
+  private boolean init;
 
   private void putKV(String key, String value, long startTS, long commitTS) {
     Mutation m =
@@ -262,14 +263,23 @@ public class LockResolverTest {
     try {
       session = TiSession.create(conf);
       pdClient = PDClient.create(session);
+      init = true;
     } catch (Exception e) {
       logger.warn("TiDB cluster may not be present");
-      // ignore npe since this test requires tidb cluster being present.
+      init = false;
     }
+  }
+
+  private void skipTest() {
+    logger.warn("Test skipped due to failure in initializing pd client.");
   }
 
   @Test
   public void getSITest() {
+    if (!init) {
+      skipTest();
+      return;
+    }
     session.getConf().setIsolationLevel(IsolationLevel.SI);
     putAlphabet();
     prepareAlphabetLocks();
@@ -295,6 +305,10 @@ public class LockResolverTest {
 
   @Test
   public void getRCTest() {
+    if (!init) {
+      skipTest();
+      return;
+    }
     session.getConf().setIsolationLevel(IsolationLevel.RC);
     putAlphabet();
     prepareAlphabetLocks();
@@ -304,6 +318,10 @@ public class LockResolverTest {
 
   @Test
   public void cleanLockTest() {
+    if (!init) {
+      skipTest();
+      return;
+    }
     session.getConf().setIsolationLevel(IsolationLevel.SI);
     for (int i = 0; i < 26; i++) {
       String k = String.valueOf((char) ('a' + i));
@@ -354,6 +372,10 @@ public class LockResolverTest {
 
   @Test
   public void txnStatusTest() {
+    if (!init) {
+      skipTest();
+      return;
+    }
     session.getConf().setIsolationLevel(IsolationLevel.SI);
     TiTimestamp startTs = pdClient.getTimestamp(backOffer);
     TiTimestamp endTs = pdClient.getTimestamp(backOffer);
@@ -402,6 +424,10 @@ public class LockResolverTest {
 
   @Test
   public void SITest() {
+    if (!init) {
+      skipTest();
+      return;
+    }
     session.getConf().setIsolationLevel(IsolationLevel.SI);
     TiTimestamp startTs = pdClient.getTimestamp(backOffer);
     TiTimestamp endTs = pdClient.getTimestamp(backOffer);
@@ -439,6 +465,10 @@ public class LockResolverTest {
 
   @Test
   public void RCTest() {
+    if (!init) {
+      skipTest();
+      return;
+    }
     session.getConf().setIsolationLevel(IsolationLevel.RC);
     TiTimestamp startTs = pdClient.getTimestamp(backOffer);
     TiTimestamp endTs = pdClient.getTimestamp(backOffer);

--- a/tikv-client/src/test/resources/log4j.properties
+++ b/tikv-client/src/test/resources/log4j.properties
@@ -1,2 +1,9 @@
-log4j.rootLogger=DEBUG
+# Set root logger level to DEBUG and its only appender to A1.
+log4j.rootLogger=INFO, A1
 
+# A1 is set to be a ConsoleAppender.
+log4j.appender.A1=org.apache.log4j.ConsoleAppender
+
+# A1 uses PatternLayout.
+log4j.appender.A1.layout=org.apache.log4j.PatternLayout
+log4j.appender.A1.layout.ConversionPattern=%-4r [%t] %-5p %c %x - %m%n

